packages/extension-base/src/hooks/use-extension-config.ts

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {useApplicationExtension} from '@salesforce/pwa-kit-extension-sdk/react'
+import extensionMeta from '../../extension-meta.json'
+
+/**
+ * This hook returns the configuration for the current application extension.
+ */
+export const useExtensionConfig = () => {
+    const extension = useApplicationExtension(extensionMeta.id)
+    if (extension === undefined) {
+        throw new Error(`'useExtensionConfig' could not find your current application extension instance!`)
+    }
+    return extension?.getConfig()
```

Comment:
At this point, we already checked that the extension is defined.

```suggestion
    return extension.getConfig()
```